src/Microsoft.AspNetCore.Mvc/HostingApplicationNameProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    public class HostingApplicationNameProvider : IApplicationNameProvider
+    {
+        private IHostingEnvironment _environment;
+
+        public HostingApplicationNameProvider(IServiceCollection services)
```

Comment:
I'm probably missing something here, but why doesn't this take `IHostingEnvironment` as a ctor param and resolve it from DI the "normal" way?